wallet/wallet-cli-lib/src/commands/mod.rs

```diff
@@ -231,382 +246,501 @@ fn print_coin_amount(chain_config: &ChainConfig, value: Amount) -> String {
+                    }
+                    None => wallet_controller::mnemonic::generate_new_mnemonic(language),
+                };
+
+                let wallet = RpcController::create_wallet(
+                    Arc::clone(chain_config),
+                    wallet_path,
+                    mnemonic.clone(),
+                    None,
+                )
+                .map_err(WalletCliError::Controller)?;
+
+                let number_of_accounts = wallet.number_of_accounts();
+                *controller_opt = Some(RpcController::new(
+                    Arc::clone(chain_config),
+                    rpc_client.clone(),
+                    wallet,
+                ));
+
+                let msg = if need_mnemonic_backup {
+                    format!(
+                    "New wallet created successfully\nYour mnemonic: {}\nPlease write it somewhere safe to be able to restore your wallet."
+                , mnemonic)
+                } else {
+                    "New wallet created successfully".to_owned()
+                };
+                self.set_total_accounts(number_of_accounts);
+                Ok(ConsoleCommand::SetStatus {
+                    status: self.repl_status(),
+                    print_message: msg,
+                })
             }
 
-            Ok(ConsoleCommand::Print("Success".to_owned()))
-        }
+            WalletCommand::OpenWallet { wallet_path } => {
+                utils::ensure!(
+                    controller_opt.is_none(),
+                    WalletCliError::WalletFileAlreadyOpen
+                );
+
+                let wallet = RpcController::open_wallet(Arc::clone(chain_config), wallet_path)
+                    .map_err(WalletCliError::Controller)?;
+
+                let number_of_accounts = wallet.number_of_accounts();
+                *controller_opt = Some(RpcController::new(
+                    Arc::clone(chain_config),
+                    rpc_client.clone(),
+                    wallet,
+                ));
+
+                self.set_total_accounts(number_of_accounts);
+                Ok(ConsoleCommand::SetStatus {
+                    status: self.repl_status(),
+                    print_message: "Wallet loaded successfully".to_owned(),
+                })
+            }
 
-        WalletCommand::RemovePrivateKeysEncryption => {
-            match controller_opt.as_mut() {
-                None => {
-                    return Err(WalletCliError::NoWallet);
-                }
-                Some(controller) => {
-                    controller.encrypt_wallet(&None).map_err(WalletCliError::Controller)?;
+            WalletCommand::CloseWallet => {
+                utils::ensure!(controller_opt.is_some(), WalletCliError::NoWallet);
+
+                *controller_opt = None;
+
+                self.state = None;
+                Ok(ConsoleCommand::SetStatus {
+                    status: self.repl_status(),
+                    print_message: "Success".to_owned(),
```

Comment:
Should this and the following "Success" messages output more context like the ones above e.g. "Wallet was successfully closed"? Otherwise, multiple lines of "Success" can get a bit confusing.

Reply:
I think it's OK because you can see the command that produced the output. It should be clear enough.